src/app/pages/products-list/products-list.component.html

```diff
@@ -23,22 +20,53 @@
     <app-card class="card" [product]="productsFromNgIf[3]" (buy)="onProductBuy($event)"></app-card>
     <app-card class="card" [product]="productsFromNgIf[4]" (buy)="onProductBuy($event)"></app-card> -->
 
-    <app-card
-        *ngFor="
-            let product of productsFromNgIf;
-            trackBy: trackById;
-            let index = index;
-            let first = first;
-            let last = last;
-            let even = even;
-            let odd = odd
+    <!-- Раскомментрировать для ДЗ -->
+    <ng-container
+        *appPagination="
+            let productsGroup of products;
+            chankSize: 2;
+            let pageIndexes = pageIndexes;
+            let activeIndex = activeIndex;
+            let next = next;
+            let back = back;
+            let selectIndex = selectIndex
         "
-        class="card"
-        [product]="product"
-        (buy)="onProductBuy($event)"
-    ></app-card>
-</div>
-<!-- </ng-template> -->
+    >
+        <div class="cards">
+            <app-card
+                *ngFor="
+                    let product of productsGroup;
+                    trackBy: trackById;
+                    let index = index;
+                    let first = first;
+                    let last = last;
+                    let even = even;
+                    let odd = odd
+                "
+                class="card"
+                [product]="product"
+                (buy)="onProductBuy($event)"
+            ></app-card>
+        </div>
+        <div class="pagination">
+            <button type="button" (click)="back(); $event.stopPropagation()">назад</button>
+            <div>
+                <span>
+                    <button
+                        *ngFor="let pageIndex of pageIndexes"
+                        type="button"
+                        [ngClass]="{'active-page': pageIndex === activeIndex}"
```

Comment:
Мысли в слух: мне нравится другой синтаксис, считаю его более лаконичным и элегантным:
```suggestion
                        [class.active-page]="pageIndex === activeIndex"
```

Reply:
исправил